cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -317,6 +330,15 @@ List<ProgramSpecification> getDeletedProgramSpecifications(ApplicationId id,
   @Nullable
   ApplicationSpecification getApplication(ApplicationId id);
 
+  /**
+   * Returns application metadata information by id.
+   *
+   * @param id application id
+   * @return application metadata
+   */
+  @Nullable
+  ApplicationMeta getApplicationMetadata(ApplicationId id);
```

Comment:
We want this function so that we return the newly added info - changsummary, author and creation time in the response

Reply:
yes, this will need refactoring in several places. I've created https://cdap.atlassian.net/browse/CDAP-19644 to track

Reply:
So this is for getting the application meta for a specific application version?

Reply:
yes